### ISSUE TITLE
Direct deployment via GitHub Actions

### ISSUE THREAD

**Issue by simonbyrne** [JuliaDocs/Documenter.jl#1959]
GitHub Pages are now deployed via GitHub Actions, see https://github.blog/2022-08-10-github-pages-now-uses-actions-by-default/ for more details. You can see the current Action which does this for Documenter.jl here:
https://github.com/JuliaDocs/Documenter.jl/actions/workflows/pages/pages-build-deployment

This means that instead of Documenter pushing to the `gh-pages` branch, it should be possible to deploy directly via GitHub Actions. It's not clear how this would work for keeping versioned docs around (would it still need a branch to keep track of this?)

**Comment by simonbyrne** [comment]
It's a bit buried in the docs, but there are some more details here: https://docs.github.com/en/pages/getting-started-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site#creating-a-custom-github-actions-workflow-to-publish-your-site

**Comment by mortenpi** [comment]
This is interesting. But one issue I see with it is that I think uploaded artifact replaces the whole Pages contents (like GitLab)? And I didn't see any way to access the existing files? Documenter updates the `gh-pages` contents incrementally on every build (so that you'd have the old version docs available too).

This is probably less of an issue, but you'll also lose the history on the `gh-pages` branch, which can be useful when debugging, or restoring things of something goes very wrong.

**Comment by simonbyrne** [comment]
Yeah, in that case I'm not sure it's worth it? To be honest, I only reason I noticed this at all is because I was updating MPI.jl which has a huge CI matrix, and that was blocking the deployment jobs from being run. One thing we could do is move the deployment to be an extra explicit step in the documenter action though (without changing the current workflow)
